### PR TITLE
Add a TotalThroughputSampler.

### PR DESCRIPTION
**Pull request by magnusstahre**

### CODE REVIEW

**Comment by maplebed** [comment]
This looks great. Thank you for bringing it in!

File: sample/totalthroughput.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		d.Metrics.IncrementCounter("dynsampler_num_kept")
+	} else {
+		d.Metrics.IncrementCounter("dynsampler_num_dropped")
+	}
+	d.Metrics.Histogram("dynsampler_sample_rate", float64(rate))
+	return uint(rate), shouldKeep
+}
+
+// buildKey takes a trace and returns the key to use for the dynsampler.
+func (d *TotalThroughputSampler) buildKey(trace *types.Trace) string {
```
Comment (martin308) [inline sample/totalthroughput.go:95]: once we get #186 merged we can share the code to generate the key!

**Review by martin308 — COMMENTED** [review]
Looks good, thanks @magnusstahre, once we get #186 in then we can refactor this to use the new shared key generation, and once #188 is merged the build should pass!

**Comment by martin308** [comment]
@magnusstahre I'm also happy to make the above changes to your PR if you like?

File: sample/totalthroughput.go (hunk 1 above)
Comment (magnusstahre) [reply sample/totalthroughput.go]: @martin308 my PR has been rebased to leverage #186 changes

**Review by martin308 — CHANGES_REQUESTED** [review]
One more thing I missed, sorry! Copied over some of the validation from the other dynamic samples that look like they matched up

**Review by martin308 — APPROVED** [review]

**Comment by magnusstahre** [comment]
@martin308 are you able to cut a release that has this change in it? thanks!

**Comment by MikeGoldsmith** [comment]
Hey @magnusstahre - I've just published [v0.13.0](https://github.com/honeycombio/refinery/releases/tag/v0.13.0) 👍

**Pull request merged**